SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -2065,17 +2068,22 @@ void CSolver::AdaptCFLNumber(CGeometry **geometry,
     /* Check whether we achieved the requested reduction in the linear
      solver residual within the specified number of linear iterations. */
 
-    bool reduceCFL = false;
-    su2double linResFlow = solverFlow->GetResLinSolver();
-    su2double linResTurb = -1.0;
-    if ((iMesh == MESH_0) && (config->GetKind_Turb_Model() != NONE)) {
-      linResTurb = solverTurb->GetResLinSolver();
-    }
+    su2double linResTurb = 0.0;
+    if ((iMesh == MESH_0) && solverTurb) linResTurb = solverTurb->GetResLinSolver();
 
-    su2double maxLinResid = max(linResFlow, linResTurb);
-    if (maxLinResid > 0.5) {
-      reduceCFL = true;
-    }
+    unsigned short linIterTurb = 0;
+    if ((iMesh == MESH_0) && solverTurb) linIterTurb = solverTurb->GetIterLinSolver();
+
+    /* Max residual and iterations between flow and turbulence. */
+    const su2double linRes = max(solverFlow->GetResLinSolver(), linResTurb);
+    const auto linIter = max(solverFlow->GetIterLinSolver(), linIterTurb);
+
+    /* Tolerance limited to a "reasonable" value. */
+    const su2double linTol = max(0.001, config->GetLinear_Solver_Error());
```

Comment:
To work around the cases that have unreasonably low tolerances I added this 0.001.

Reply:
Might be a good idea to mention this limit in the config template